netcdf 4.6.0 with opendap capabilities

pmjherman opened the pull request:
Until now, there was no MINGW build of netcdf with the capability of opendap access to remote thredds servers. The only possibility in Windows was to use a binary produced with Microsoft Visual C by the people issuing netcdf. This prevented, however, incorporation into R and maybe other MINGW-based software.

In this build, I have patched the netcdf software to allow for opendap, and changed the curl build slightly in order to make a build possible.

pmjherman commented:
The build of continuous-integration/tea failed because it did not have a cmake version that was recent enough. It does work, however, with a recent MSYS2 system where cmake is up to date.
I have tested the build on a multitude of netcdf and opendap files and found no problem. I have also used the library successfully to build an R package.

Alexpux commented on mingw-w64-netcdf/PKGBUILD:
```diff
@@ -29,13 +46,17 @@ build() {
     -DBUILD_TESTING=OFF \
     -DENABLE_DYNAMIC_LOADING=OFF \
     -DENABLE_NETCDF_4=ON \
+    -DZLIB_LIBRARY:FILEPATH=${MINGW_PREFIX}/lib/libz.a \
```
I will not accept forcing static linking. We use shared linking by default

Alexpux commented on mingw-w64-netcdf/PKGBUILD:
```diff
@@ -44,13 +65,18 @@ build() {
     -DENABLE_DYNAMIC_LOADING=OFF \
     -DENABLE_NETCDF_4=ON \
+	-DENABLE_LOGGING=ON \
+    -DZLIB_LIBRARY:FILEPATH=${MINGW_PREFIX}/lib/libz.a \
```
The same

Alexpux commented on mingw-w64-netcdf/PKGBUILD:
```diff
@@ -44,13 +65,18 @@ build() {
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
   ${MINGW_PREFIX}/bin/cmake -G "MSYS Makefiles" \
-    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+	 -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
```
Dont use TAB's

Alexpux commented on mingw-w64-netcdf/PKGBUILD:
```diff
@@ -29,13 +46,17 @@ build() {
   ${MINGW_PREFIX}/bin/cmake -G "MSYS Makefiles" \
     -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
-    -DCMAKE_BUILD_TYPE=Release \
+	-DCMAKE_BUILD_TYPE=Release \
```
Dont use TABs

Alexpux commented:
@pmjherman fix my comments

pmjherman commented:
Thanks for the review and sorry for my beginner's mistakes. I have changed the editorial things (TABs). I found out that the compiler links to the libs could be left out alltogether. I added them with a previous version of netcdf, I think they may have been needed because of a bug in the cmake system at that time. Anyway, this also solves the problem of static linking. I hope that with this all will be OK.
I will commit my changes and then give a new pull request

Alexpux commented:
Rebase changes on top of cerrent state. Also netcdf 4.6.1 releaseed

Alexpux commented:
@pmjherman ping?

Alexpux commented:
Fixed by https://github.com/Alexpux/MINGW-packages/commit/47e5d7d4c15510c3d092a6dfefd8a109bc03b78a
